### PR TITLE
DAOS-15038 test: Parsing NVMe controller details from SMD info JSON is failing

### PR DESCRIPTION
**Pull request by tanabarr**
JSON output format of dmg storage query list-devices changed in commit
8e1c8468428e7b2ac97975300e18d0aaff8e381c and specifically dev_state
and health_stats were moved to a ctrlr sub-node of smd_info. This has
broken parse_device_info() and dmg_storage_query_device_health() in
src/common/tests_dmg_helpers.c. This change updates the function to
look for dev_state and health_stats keys inside the ctrlr sub-node.

Also fix NVMe recovery test 4 by removing check for non-existent error
message in control log as noted in DAOS-14907.

Features: DaosCoreTestNvme
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — DISMISSED** [review]

**Review by NiuYawei — DISMISSED** [review]

**Review by knard-intel — DISMISSED** [review]

File: src/common/tests_dmg_helpers.c [hunk 1]
```diff
@@ -1277,9 +1283,10 @@ dmg_storage_query_device_health(const char *dmg_config_file, char *host,
 	struct json_object	*storage_map = NULL;
 	struct json_object	*smd_info = NULL;
 	struct json_object	*storage_info = NULL;
-	struct json_object	*health_info = NULL;
+	struct json_object       *health_stats = NULL;
```
Comment (knard-intel) [inline src/common/tests_dmg_helpers.c:1286]: NIT indentation issue

File: src/common/tests_dmg_helpers.c [hunk 2]
```diff
@@ -1277,9 +1283,10 @@ dmg_storage_query_device_health(const char *dmg_config_file, char *host,
...
 	struct json_object	*devices = NULL;
 	struct json_object	*dev_info = NULL;
+	struct json_object       *ctrlr_info   = NULL;
```
Comment (knard-intel) [inline src/common/tests_dmg_helpers.c:1289]: NIT indentation issue

**Review by knard-intel — DISMISSED** [review]
LGTM

**Review by NiuYawei — APPROVED** [review]

**Comment by tanabarr** [comment]
GATEKEEPER: Please use the PR title and description above as the commit message when landing. TIA

**Review by kjacque — DISMISSED** [review]
LGTM

**Review by daltonbohning — DISMISSED** [review]

File: src/common/tests_dmg_helpers.c (hunk 1 above)
Comment (tanabarr) [reply src/common/tests_dmg_helpers.c:1286]: I will fix these either if I repush or in a subsequent PR

File: src/common/tests_dmg_helpers.c (hunk 2 above)
Comment (tanabarr) [reply src/common/tests_dmg_helpers.c:1289]: I will fix these either if I repush or in a subsequent PR

**Comment by tanabarr** [comment]
In CI run https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13630/4/tests/ the unit tests failed as expected due to breakage that has been addressed by the landing of PR-13672. NVMe recovery tests have been fixed but there is an unrelated known failure on REBUILD15 - DER_BUSY DAOS-14682.

@phender @NiuYawei do you think it's feasible requesting forced landing given the above test results or should I rebase and repush?

**Review by knard-intel — APPROVED** [review]

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**